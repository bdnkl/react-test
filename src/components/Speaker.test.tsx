import React from "react";
import Speaker from "./Speaker";
import renderer from "react-test-renderer"

describe("speaker component", () => {
  const testSpeaker = {
    id: "1",
    bio: "bio",
    first: "Jane",
    last: "Doe",
    company: "GitHub",
    twitterHandle: "janedoe",
    favorite: true,
    sessions: [{id: "1", title: "session title", room: {name: "room", capacity: 10}, eventYear: "2022"}]
  }

  it("makes sure the speaker component renders correctly", () => {
    const tree = renderer.create(
      <Speaker speaker={testSpeaker}/>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
