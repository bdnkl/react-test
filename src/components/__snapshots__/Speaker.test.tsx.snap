// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`speaker component should make sure the speaker component renders correctly 1`] = `
<div
  className="col-xs-12 col-sm-12 col-md-6 col-lg-4 col-sm-12 col-xs-12"
>
  <div
    className="card card-height p-4 mt-4"
  >
    <div
      className="speaker-img d-flex flex-row justify-content-center align-items-center h-300"
    >
      <img
        alt="Jane Doe"
        className="contain-fit"
        src="/images/speaker-1.jpg"
        width="300"
      />
    </div>
    <div
      className="speaker-info"
    >
      <div
        className="d-flex justify-content-between mb-3"
      >
        <h3
          className="text-truncate w-200"
        >
          Jane
           
          Doe
        </h3>
      </div>
      <div>
        <p
          className="card-description"
        >
          bio
        </p>
        <div
          className="social d-flex flex-row mt-4"
        >
          <div
            className="company"
          >
            <h5>
              Company
            </h5>
            <h6>
              GitHub
            </h6>
          </div>
          <div
            className="twitter"
          >
            <h5>
              Twitter
            </h5>
            <h6>
              janedoe
            </h6>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sessionBox card h-250"
  >
    <span
      className="session w-100"
    >
      session title
       
      <strong>
        Room: 
        room
      </strong>
    </span>
  </div>
</div>
`;
